### ISSUE TITLE
amdgpu-utils-env is NOT available

### ISSUE THREAD

**Issue by Mistico83** [Ricks-Lab/gpu-utils#110]
Firstly, I do apologise if I say something stupid.

amdgpu-chk says:

`Using python 3.8.5
           Python version OK. 
Using Linux Kernel 5.9.1-050901-lowlatency
           OS kernel OK. 
AMD GPU driver is driver=amdgpu latency=0
           AMD driver OK. 
python3 venv is installed
           python3-venv OK. 
amdgpu-utils-env is NOT available
           amdgpu-utils-env should be configured per User Guide. 
Environment not configured. WARNING
Not in amdgpu-utils-env
           amdgpu-utils-env should be activated per User Guide. 
Virtual Environment not activated. WARNING`

I had edited grub:

`GRUB_CMDLINE_LINUX_DEFAULT="quiet splash radeon.si_support=0 radeon.cik_support=0 amdgpu.si_support=1 amdgpu.cik_support=1 amdgpu.dc=1 amdgpu.ppfeaturemask=0xffff7fff"`

I try to run amdgpu-ls, amdgpu-monitor, amdgpu-pac, amdgpu-plot. Same result:

`AMD Wattman features enabled: 0xffff7fff
amdgpu version: UNKNOWN
No AMD GPUs detected, exiting...`

What I'm doing wrong?

Maybe you want to know what lshw says:

`  *-display                 
       description: VGA compatible controller
       product: Pitcairn XT [Radeon HD 7870 GHz Edition]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:01:00.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom
       configuration: driver=amdgpu latency=0
       resources: irq:42 memory:c0000000-cfffffff memory:ff600000-ff63ffff ioport:e000(size=256) memory:c0000-dffff`

Basic system details ahead:

`NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal`

Kernel is Linux 5.9.1-050901-lowlatency

I had disabled the Radeon HD 7660D IGP (onchip in mainboard) because your program detects first radeon driver (whilst is blacklisted) and no the PCI-e card. Does it support multicards?

I'm lost and mentally exhausted. I've read user guide and readme, but I don't found the point. Thank you.

**Comment by Ricks-Lab** [comment]
It looks like you installed from debian package repository.  I am working an issue with updating that package which stopped when I changed the name to rickslab-gpu-utils.  There are two ways to install the latest.  But first you need to remove ricks-amdgpu-utils:
```
sudo apt purge ricks-amdgpu-utils
sudo apt autoremove
```
Then install either from PyPI or from the debian package on rickslab.com.  Instructions for both are in the latest Readme file in this repository.  In later versions, it is made clear that setting up a virtual env is only required if you wish to replicate the development environment in order to contribute to the project.  The latest gpu-chk is more clear about this.

Let me know if you have any issues.  Thanks!

**Comment by Mistico83** [comment]
Thank you for the quick response. Yes, I've installed it as an Ubuntu package (I prefer the distributed packages for stability).

I could install the last version following your instructions. But I have another problem, don't know if I have to open another issue.

gpu-chk says:

```
Using python 3.8.5
           Python version OK. 
Using Linux Kernel 5.9.1-050901-lowlatency
           OS kernel OK. 
Using Linux distribution: Ubuntu 20.04.2 LTS
           Distro has been Validated. 
amdgpu/rocm version: UNKNOWN
           gpu-utils can still be used. 
python3 venv is installed
           python3-venv OK. 
rickslab-gpu-utils-env is NOT available
           rickslab-gpu-utils-env can be configured per User Guide. 
Virtual Environment not configured. Only required by developers.
Not in rickslab-gpu-utils-env (Only needed if you want to duplicate dev env)
          rickslab-gpu-utils-env can be activated per User Guide.
```
And the other commands:

```
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xffff7fff
Warning: Can not read parameter: loading, disabling for this GPU: 0
Warning: Can not read parameter: mem_loading, disabling for this GPU: 0
Traceback (most recent call last):
  File "/usr/bin/gpu-ls", line 150, in <module>
    main()
  File "/usr/bin/gpu-ls", line 121, in main
    gpu_list.read_gpu_sensor_set(data_type=Gpu.GpuItem.SensorSet.All)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 2106, in read_gpu_sensor_set
    gpu.read_gpu_sensor_set(data_type)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 1235, in read_gpu_sensor_set
    return self.read_gpu_sensor_set_amd(data_type)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 1372, in read_gpu_sensor_set_amd
    self.set_params_value(param, rdata)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 576, in set_params_value
    self.sclk_dpm_state.update({int(sclk_ps[0]): sclk_ps[1]})
ValueError: invalid literal for int() with base 10: ''
```
Any idea?

**Comment by Ricks-Lab** [comment]
Can you run with `--debug` option and post the resultant log file here?

**Comment by Ricks-Lab** [comment]
I pushed a version that addresses this specific issue, but concerned that the older cards capability is not being assessed correctly.  The log output will help to determine that. 

You can use the latest in the repository by cloning it and running the `./gpu-ls` from that directory.

**Comment by Ricks-Lab** [comment]
I installed an old Pitcairn XT [Radeon HD 7870 GHz Edition] GPU I had, and found no issues on my system.  The output of lshw is identical to that of your card.  Perhaps there is something different in the way it responds in the kernel version that you are using.  Having the debug out as described above would be very helpful in understanding the issue.  Thanks!

**Comment by Ricks-Lab** [comment]
I have just tried with both kernels 5.9 and 5.9.1 (generic) with the same GPU as yours and found that `gpu-ls` works as expected.

Are you certain that you have removed `ricks-amdgpu-utils`? If it is still installed when installing `rickslab-gpu-utils` it is possible that `gpu-ls` will get the old module.  You should execute `sudo purge ricks-amdgpu-utils` and `sudo apt autoremove`

**Comment by Mistico83** [comment]
I tried to reinstall, is that correct?

```
sudo apt purge ricks-amdgpu-utils
sudo apt autoremove
wget -q -O - https://debian.rickslab.com/PUBLIC.KEY | sudo apt-key add -
echo 'deb [arch=amd64] https://debian.rickslab.com/gpu-utils/ eddore main' | sudo tee /etc/apt/sources.list.d/rickslab-gpu-utils.list
sudo apt update
sudo apt install rickslab-gpu-utils
```
Trying gpu-chk --debug
```
usage: gpu-chk [-h] [--about]
gpu-chk: error: unrecognized arguments: --debug
```
gpu-ls --debug
```
Ubuntu: Validated
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xffff7fff
Warning: Can not read parameter: loading, disabling for this GPU: 0
Warning: Can not read parameter: mem_loading, disabling for this GPU: 0
Traceback (most recent call last):
  File "/usr/bin/gpu-ls", line 150, in <module>
    main()
  File "/usr/bin/gpu-ls", line 121, in main
    gpu_list.read_gpu_sensor_set(data_type=Gpu.GpuItem.SensorSet.All)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 2106, in read_gpu_sensor_set
    gpu.read_gpu_sensor_set(data_type)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 1235, in read_gpu_sensor_set
    return self.read_gpu_sensor_set_amd(data_type)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 1372, in read_gpu_sensor_set_amd
    self.set_params_value(param, rdata)
  File "/usr/lib/python3/dist-packages/GPUmodules/GPUmodule.py", line 576, in set_params_value
    self.sclk_dpm_state.update({int(sclk_ps[0]): sclk_ps[1]})
ValueError: invalid literal for int() with base 10: ''
```
Next step?

**Comment by Ricks-Lab** [comment]
There should be a debug log file in the directory from where you executed the command. Please post the contents of the file here.

**Comment by Mistico83** [comment]
Here you are.
[debug_gpu-utils_20210311-120843.log](https://github.com/Ricks-Lab/gpu-utils/files/6122599/debug_gpu-utils_20210311-120843.log)

**Comment by Ricks-Lab** [comment]
Thanks for providing the debug log file.  It is clear now that the card is being classified on your system as an APU while my system is classifying it as LEGACY.  I have pushed a version with additional debug statement.  Can you clone the update repository.  Then cd into your clone of the repository and execute `./gpu-ls --debug`.  And then post the debug log?  Thanks!

**Comment by Ricks-Lab** [comment]
I think I found the issue.  For some reason, your system driver files include `pp_dpm_mclk` and `pp_dpm_sclk` files which did not exist on my system.  They don't return any data when queried on your system.  I have removed the check for these files in determining the type as APU.  Latest is in the repository.  I will create a package once we can confirm the problem is fixed.

**Comment by Mistico83** [comment]
Here you are the latest debug log from updated repository via git clone.
[debug_gpu-utils_20210312-132812.log](https://github.com/Ricks-Lab/gpu-utils/files/6129994/debug_gpu-utils_20210312-132812.log)

**Comment by Ricks-Lab** [comment]
Looks like you are still running the previous release:
```
DEBUG:gpu-utils:gpu-ls.main:########## gpu-ls 3.5.6
```
To run from the repository, you should cd into the directory of the cloned repository and execute:
```
./gpu-ls --debug
```

**Comment by Mistico83** [comment]
Forgive me, I've executed the command without ./

Here's the updated debug log.
[debug_gpu-utils_20210312-134621.log](https://github.com/Ricks-Lab/gpu-utils/files/6130096/debug_gpu-utils_20210312-134621.log)

**Comment by Ricks-Lab** [comment]
Now the log file indicates the correct GPU type as Legacy.  Does it display results without error now?

**Comment by Ricks-Lab** [comment]
This is the output from my system:
```
Ubuntu: Validated
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xfffd7fff
1 total GPUs, 0 rw, 1 r-only, 0 w-only

Card Number: 0
   Vendor: AMD
   Readable: True
   Writable: False
   Compute: False
   GPU UID: None
   Device ID: {'device': '0x6818', 'subsystem_device': '0x2740', 'subsystem_vendor': '0x1462', 'vendor': '0x1002'}
   Decoded Device ID: Pitcairn XT [Radeon HD 7870 GHz Edition]
   Card Model: Advanced Micro Devices, Inc. [AMD/ATI] Pitcairn XT [Radeon HD 7870 GHz Edition]
   Display Card Model: Pitcairn XT [Radeon HD 7870 GHz Edition]
   PCIe ID: 08:00.0
      Link Speed: 8.0 GT/s PCIe
      Link Width: 16
   ##################################################
   Driver: radeon, amdgpu
   Compute Platform: None
   GPU Type: Legacy
   HWmon: /sys/class/drm/card0/device/hwmon/hwmon0
   Card Path: /sys/class/drm/card0/device
   System Card Path: /sys/devices/pci0000:00/0000:00:03.1/0000:08:00.0
   ##################################################
   Fan PWM Mode: [2, 'Dynamic']
   Current Fan PWM (%): 28
      Fan PWM Range (%): [0, 100]
   ##################################################
   Current  Temps (C): {'temp1_input': 31.0}
   Critical Temps (C): {'temp1_crit': 120.0}
   Power DPM Force Performance Level: auto
```

**Comment by Mistico83** [comment]
gpu-chk says:

```
Using python 3.8.5
           Python version OK. 
Using Linux Kernel 5.9.1-050901-lowlatency
           OS kernel OK. 
Using Linux distribution: Ubuntu 20.04.2 LTS
           Distro has been Validated. 
amdgpu/rocm version: UNKNOWN
           gpu-utils can still be used. 
python3 venv is installed
           python3-venv OK. 
rickslab-gpu-utils-env is NOT available
           rickslab-gpu-utils-env can be configured per User Guide. 
Virtual Environment not configured. Only required by developers.
Not in rickslab-gpu-utils-env (Only needed if you want to duplicate dev env)
          rickslab-gpu-utils-env can be activated per User Guide. 
```
and gpu-ls says:
```
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xffff7fff
1 total GPUs, 0 rw, 1 r-only, 0 w-only

Card Number: 0
   Vendor: AMD
   Readable: True
   Writable: False
   Compute: False
   GPU UID: None
   Device ID: {'device': '0x6818', 'subsystem_device': '0xe241', 'subsystem_vendor': '0x174b', 'vendor': '0x1002'}
   Decoded Device ID: Pitcairn XT [Radeon HD 7870 GHz Edition]
   Card Model: Advanced Micro Devices, Inc. [AMD/ATI] Pitcairn XT [Radeon HD 7870 GHz Edition]
   Display Card Model: Pitcairn XT [Radeon HD 7870 GHz Edition]
   PCIe ID: 01:00.0
      Link Speed: 5.0 GT/s PCIe
      Link Width: 16
   ##################################################
   Driver: radeon, amdgpu
   Compute Platform: None
   GPU Type: Legacy
   HWmon: /sys/class/drm/card0/device/hwmon/hwmon0
   Card Path: /sys/class/drm/card0/device
   System Card Path: /sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0
   ##################################################
   Fan PWM Mode: [0, 'None']
   Current Fan PWM (%): 35
      Fan PWM Range (%): [0, 100]
   ##################################################
   Current  Temps (C): {'edge': 42.0}
   Critical Temps (C): {'edge': 120.0}
   Power DPM Force Performance Level: auto
```
gpu-mon says:
```
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xffff7fff
1 total GPUs, 0 rw, 1 r-only, 0 w-only

No readable and compatible GPUs detected, exiting...
```
gpu-pac and gpu-plot says:
```
Detected GPUs: AMD: 1
amdgpu/rocm version: UNKNOWN
AMD: Wattman features enabled: 0xffff7fff
1 total GPUs, 0 rw, 1 r-only, 0 w-only

None are writable, exiting...
```

**Comment by Ricks-Lab** [comment]
The output is as expected.  AMD did not implement useful capability in the Linux drivers until R9 series.

**Comment by Mistico83** [comment]
What about the rickslab-gpu-utils-env, is it only util for developers?

So, It's not a matter of your code, there's a matter of the Linux drivers from AMD.

Do you think there's a remedy for our Radeon HD 7850 in Linux, or I must switch to Windows to use it at the 100% of its capabilities?

**Comment by Ricks-Lab** [comment]
The `gpu-ls` command indicates where all of the driver files and hwmon files are.  In your case, the location is:
```
HWmon: /sys/class/drm/card0/device/hwmon/hwmon0
Card Path: /sys/class/drm/card0/device
```
If you `sudo cat` all of the files, you will see all of the information that is available.  This capability is determined by the driver, not my utility.  In order to write to the GPU for overclocking, the file `pp_od_clk_voltage` must exist and be writable.   As seen in your log file (and mine), this doesn't exist for the 7870: 
```
The file /sys/class/drm/card0/device/pp_od_clk_voltage does not exist
```
I have only ever tried the Linux default drivers for AMD and the AMD drivers for Linux.  Not sure if there are other driver options that enable this for older GPUs.

**Comment by Mistico83** [comment]
OK, thank you. If I can help in other issue, contact me.

**Comment by Ricks-Lab** [comment]
Thanks for your help in troubleshooting this issue!  Still not sure why your system behaved different from mine, but I did find a flaw in the determination logic.  I will release after I do a bit more verification tomorrow.

**Comment by Ricks-Lab** [comment]
Just released v3.5.8 to PyPI.  Will wait for feedback before I release a debian package.
